backend/src/test/java/com/google/minicrm/data/FormTest.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.minicrm.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.User;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Provides Unit tests for the Form data object.
+ */
+@RunWith(JUnit4.class)
+public final class FormTest {
+
+  private static final String TEST_USER_ID = "testUserId";
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown(){
+    helper.tearDown();
+  }
+
+  @Test
+  public void form_convertToEntityAndBack() throws Exception {
+    User testUser = new User("email", "authDomain", TEST_USER_ID);
```

Comment:
For test clarity, separate into three separate blocks for "Act", "Arrange", "Assert".

Reply:
Just to clarify:
Would "Act" be setting up parameters and making the method call and
"Arrange" be arranging/sorting the return values from the method?

Reply:
Sorry, I should've ordered these. "Arrange" is the first block, which does all the setup. Basically it gets the inputs ready for what you want to test. Following that is "Act", which is the action/function performed on these inputs. Finally, we have "Assert", which are your assert statements that you want to enforce for correctness.

Reply:
yup, that's correct.